src/nevermined/NvmApp.ts

```diff
@@ -482,6 +482,10 @@ export class NvmApp {
       serviceIndex = salesService.index
       numberCredits = salesService.attributes.main.nftAttributes.amount as bigint
 
+      const service = ddo.findServiceByType('nft-sales')
```

Comment:
it seems you have already the service (`salesService` so not sure why you need to get it from the ddo again